routes/account.rb

```diff
@@ -17,13 +17,15 @@ class Clover
         r.post "disconnect" do
           provider, uid = typecast_params.nonempty_str(["provider", "uid"])
           identities = current_account.identities
-          unless identities.length > 1
-            # YYYY: We can allow to disconnect the last omniauth provider if the user has a password
-            # But need https://github.com/jeremyevans/rodauth/pull/461 to be merged first.
+          unless identities.length > (rodauth.has_password? ? 0 : 1)
             flash[:error] = "You must have at least one login method"
             r.redirect "/account/login-method"
           end
-          if (identity = identities.find { _1.provider == provider && _1.uid == uid })
+          if provider == "password"
+            DB[:account_password_hashes].where(id: current_account.id).delete(force: true)
+            DB[:account_previous_password_hashes].where(account_id: current_account.id).delete(force: true)
```

Comment:
```suggestion
            DB[:account_password_hashes].where(id: current_account.id).delete
            DB[:account_previous_password_hashes].where(account_id: current_account.id).delete
```

`(force: true)` is only required for model datasets, not for plain datasets, since 9d81cc2a62fd3f7715dafc82281ed406c51628f9

Reply:
Cool. I didn't know that.